openedx/core/djangoapps/content/learning_sequences/api/processors/user_partition_groups.py

```diff
@@ -0,0 +1,57 @@
+# lint-amnesty, pylint: disable=missing-module-docstring
+import logging
+from datetime import datetime
+
+from django.contrib.auth import get_user_model
+from opaque_keys.edx.keys import CourseKey
+
+from common.lib.xmodule.xmodule.partitions.enrollment_track_partition_generator import (
+    create_enrollment_track_partition_with_course_id
+)
+from common.lib.xmodule.xmodule.partitions.partitions import (
+    ENROLLMENT_TRACK_PARTITION_ID,
+    UserPartition,
+)
+from common.lib.xmodule.xmodule.partitions.partitions_service import get_user_partition_groups
+
+from .base import OutlineProcessor
+
+User = get_user_model()
+log = logging.getLogger(__name__)
+
+
+class UserPartitionGroupsProcessor(OutlineProcessor):
+    """
+    Processor responsible for applying all user partition group to the outline.
+
+    Currently, this processor only exclude based on EnrollmentTrackPartitionScheme.
+    This is a significant limitation, but a step towards the goal of supporting
+    all partition schemes in the future
+    """
+
+    def _should_block_be_removed(self, block):
+        for partition_id, groups in block.user_partition_groups:
+            if partition_id == ENROLLMENT_TRACK_PARTITION_ID:
+                user_partition = create_enrollment_track_partition_with_course_id(self.course_key)
```

Comment:
Hmmm... this might actually imply that `load_data` (from `OutlineProcessor`) should take a `CourseOutlineData` as an argument, since you should only need to have to ask for the partitions that actually exist in the content.